source/lexbor/selectors/selectors.c

```diff
@@ -338,10 +338,16 @@ lxb_selectors_state_tree(lxb_selectors_t *selectors, lxb_dom_node_t *root,
 
         if (node->type == LXB_DOM_NODE_TYPE_DOCUMENT) {
             node = root->first_child;
+            if (node == NULL) {
+                goto out;
+            }
         }
     }
     else {
         node = root->first_child;
+        if (node == NULL) {
+            goto out;
+        }
     }
 
```

Comment:
@nielsdos 

I would put one check in here. It seems to be better and more reliable than making two checks.

```C
    if (selectors->options & LXB_SELECTORS_OPT_MATCH_ROOT) {
        node = root;

        if (node->type == LXB_DOM_NODE_TYPE_DOCUMENT) {
            node = root->first_child;
        }
    }
    else {
        node = root->first_child;
    }

    if (node == NULL) {
        goto out;
    }

```

What do you think about that?

Reply:
I'm fine either way. I pushed your suggestion.